### ISSUE TITLE
Validate `VertexBufferLayout::array_stride`

### ISSUE THREAD

**Issue by abey79** [gfx-rs/wgpu#6799]
### Context

The WebGPU [spec](https://gpuweb.github.io/gpuweb/#dictdef-gpuvertexbufferlayout) constraints what the array stride of a `VertexBufferLayout` might be. In particular, it forbids it to be smaller or equal than the `max(offset + size)` over all attributes:

![image](https://github.com/user-attachments/assets/d5fdc55d-e673-468b-a681-b8bbce70df53)

One might be tempted to use a stride smaller than the attributes to implement some kind of "sliding window" over a vertex buffer (I certainly was). This is however non-compliant as per above, and may variously lead to at least:
- recent chrome versions to reject your shader
- wgpu 23.0.0+ to emit incorrect metal shader (including out-of-bound array access)
- possibly other phenomenons (I have only tested mac, WebGPU, and WebGL, the latter seemingly being unaffected)

### Issue

Currently, wgpu doesn't warn/error on non-compliant stride values. Since 23.0.0,  it also silently generate erroneous metal shader when provided with non-compliant stride.

### Desired change

wgpu should error (or warn loudly) with a reference to the spec whenever a non-compliant stride is detected.

### Reference

- PR which lead to the discovery of the above with my debugging notes, in particular the offending metal shader: https://github.com/abey79/vsvg/pull/160
- PR in which is switch from a sliding window to a compliant, multiple-buffer-binding-with-offset ("zip"):  https://github.com/abey79/vsvg/pull/159 (This may inspire a work-around to anyone facing this issue.)

**Comment by sagudev** [comment]
Indeed, we are missing check somewhere around here: https://github.com/gfx-rs/wgpu/blob/6c9f93b7de42136c5c09f195fc1c4999e5962347/wgpu-core/src/device/resource.rs#L2903-L2909

EDIT: We are missing multiple checks.
